packages/api/src/hooks/useP2PAdvertCreate.ts

```diff
@@ -0,0 +1,39 @@
+import { useCallback } from 'react';
+import useMutation from '../useMutation';
+// import useInvalidateQuery from '../useInvalidateQuery';
+
+type TPayload = Parameters<ReturnType<typeof useMutation<'p2p_advert_create'>>['mutate']>[0]['payload'];
+
+/** A custom hook that creates a P2P advert. This can only be used by an approved P2P advertiser.
+ * 
+ * To create an advert, specify the following payload arguments in the `mutate` call:
+ * @example
+ *  mutate({
+        description: 'Please transfer to account number 1234',
+        type: 'buy',
+        amount: 100,
+        max_order_amount: 50,
+        min_order_amount: 20,
+        payment_method: 'bank_transfer',
+        rate: 4.25,
+    });
+ * 
+*/
+const useP2PAdvertCreate = () => {
+    // const invalidate = useInvalidateQuery();
+    const { mutate: _mutate, ...rest } = useMutation('p2p_advert_create', {
```

Comment:
i believe we need to pass data down as well.

Reply:
with @farhan-nurzi-deriv 's idea we don't to do this anymore. 
https://github.com/binary-com/deriv-app/pull/12054/files#r1418284754